store/mongo/datastore_mongo.go

```diff
@@ -306,6 +308,87 @@ func (db *DataStoreMongo) CreateIntegration(
 	return &integration, err
 }
 
+func (db *DataStoreMongo) SetIntegrationCredentials(
+	ctx context.Context,
+	integrationId uuid.UUID,
+	credentials model.Credentials,
+) error {
+	collIntegrations := db.client.Database(DbName).Collection(CollNameIntegrations)
+
+	fltr := bson.D{{
+		Key:   KeyID,
+		Value: integrationId,
+	}}
+
+	update := bson.M{
+		"$set": bson.D{
+			{
+				Key:   KeyCredentials,
+				Value: credentials,
+			},
+		},
+	}
+
+	_, err := collIntegrations.UpdateOne(ctx,
```

Comment:
Don't you need to check result.MatchedCount, and if zero return ErrObjectNotFound?

Reply:
Fixed, thank you!